solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -232,6 +233,45 @@ public void prepare(ResponseBuilder rb) throws IOException
     }
   }
 
+  private boolean allowSkipSecondGroupingStep(final SortSpec withinGroupSpecification, final SortSpec groupSort, final boolean isReranking) {
+    // Only possible if we only want one doc per group
+    final int limit =  withinGroupSpecification.getCount();
+    final int offset = withinGroupSpecification.getOffset();
+    if ( limit != 1 || offset != 0 ) {
+        log.error("group.skip.second.step=true is not compatible with group.limit (limit={}) ", limit);
+        return false;
+    }
+
+    // Within group sort must be the same as group sort because if we skip second step no sorting within group will be done.
+    if (withinGroupSpecification.getSort() !=  groupSort.getSort()) {
```

Comment:
question/thought: is _full_ equality required technically speaking or would _prefix_ equality be sufficient? e.g. if the sorting within the group was by `X,Y,Z` and the sorting of groups was by `X,Y`. that way around in terms of prefixes, i think, or maybe the other way around.

Reply:
I agree, done in c5024d0